subsys/logging/log_backend_net.c

```diff
@@ -216,10 +216,32 @@ static void panic(struct log_backend const *const backend)
 	panic_mode = true;
 }
 
+static void sync_string(const struct log_backend *const backend,
+		     struct log_msg_ids src_level, u32_t timestamp,
+		     const char *fmt, va_list ap)
+{
+	u32_t flags = LOG_OUTPUT_FLAG_LEVEL | LOG_OUTPUT_FLAG_FORMAT_SYSLOG |
+		LOG_OUTPUT_FLAG_TIMESTAMP;
+	u32_t key;
+
+	if (!net_init_done) {
+		net_init_done = true;
```

Comment:
Why setting it to true first? 

I don't see do_net_init() doing anything with net_init_done

So instead it could be:
`if (!net_init_done && do_net_init() == 0) {
    net_initi_done = true;
}`

Reply:
Could be done like that too, just copied the same check from `send_output()`. Beside if the `do_net_init()` fails, then it is probably not worth re-trying, so I made it to do the init only once.

Reply:
if net_init_done is reset to false, do_net_init() is going to be retried on another call.

Reply:
Yep, I was wrong. It tries indeed to re-init after all :)

Reply:
Ah, now I remember the logic here. We need to retry the net setup as it might be that the network is not yet ready when the logging is called, and so the setup might fail.

Reply:
yep makes sense. though the net_init_done check, set, unset could be revised.

Reply:
I updated the net_init_done check.